### PR TITLE
ref: Add ruff rules TRY3xx

### PR DESCRIPTION
**Pull request by cbornet**
Add ruff rules TRY3xx
[TRY301](https://docs.astral.sh/ruff/rules/raise-within-try/) issues are fixed but not enforcing them atm as they are a bit hard to grok.

### CODE REVIEW

File: src/backend/base/langflow/api/v1/endpoints.py [hunk 1]
```diff
@@ -434,35 +436,59 @@ async def experimental_run_flow(
...
-                msg = f"Flow {flow_id_str} not found"
-                raise ValueError(msg)
+        except sa.exc.StatementError as exc:
+            # StatementError('(builtins.ValueError) badly formed hexadecimal UUID string')
+            if "badly formed hexadecimal UUID string" in str(exc):
+                logger.exception(f"Flow ID {flow_id_str} is not a valid UUID")
+                # This means the Flow ID is not a valid UUID which means it can't find the flow
+                raise HTTPException(status_code=status.HTTP_404_NOT_FOUND, detail=str(exc)) from exc
+            logger.exception(exc)
+            raise HTTPException(status_code=status.HTTP_500_INTERNAL_SERVER_ERROR, detail=str(exc)) from exc
```
Comment (cbornet) [inline src/backend/base/langflow/api/v1/endpoints.py:469]: It seems this exception was missing in `StatementError`'s handling

File: src/backend/base/langflow/api/v1/folders.py [hunk 2]
```diff
@@ -203,12 +214,17 @@ async def download_file(
...
     """Download all flows from folder."""
     try:
-        return session.exec(select(Folder).where(Folder.id == folder_id, Folder.user_id == current_user.id)).first()
+        folder = session.exec(select(Folder).where(Folder.id == folder_id, Folder.user_id == current_user.id)).first()
     except Exception as e:
         if "No result found" in str(e):
             raise HTTPException(status_code=404, detail="Folder not found") from e
         raise HTTPException(status_code=500, detail=str(e)) from e
 
+    if not folder:
```
Comment (cbornet) [inline src/backend/base/langflow/api/v1/folders.py:223]: It seems this was missing

File: src/backend/base/langflow/api/v1/folders.py [hunk 3]
```diff
@@ -203,12 +214,17 @@ async def download_file(
 ):
     """Download all flows from folder."""
     try:
-        return session.exec(select(Folder).where(Folder.id == folder_id, Folder.user_id == current_user.id)).first()
+        folder = session.exec(select(Folder).where(Folder.id == folder_id, Folder.user_id == current_user.id)).first()
     except Exception as e:
         if "No result found" in str(e):
```
Comment (cbornet) [inline src/backend/base/langflow/api/v1/folders.py:219]: Not sure how this can happen...

File: src/backend/base/langflow/api/v1/chat.py [hunk 4]
```diff
@@ -638,70 +723,6 @@ async def build_vertex_stream(
...
-                    exc_message = "This stream has already been closed."
-                yield str(StreamData(event="error", data={"error": exc_message}))
-            finally:
-                logger.debug("Closing stream")
-                if graph:
-                    await chat_service.set_cache(flow_id_str, graph)
-                yield str(StreamData(event="close", data={"message": "Stream closed"}))
-
-        return StreamingResponse(stream_vertex(), media_type="text/event-stream")
+        return StreamingResponse(_stream_vertex(str(flow_id), vertex_id, chat_service), media_type="text/event-stream")
```
Comment (ogabrielluiz) [inline src/backend/base/langflow/api/v1/chat.py:726]: Were you able to test this? IIRC, it is a requirement to define the function inside the endpoint.

Comment (cbornet) [reply src/backend/base/langflow/api/v1/chat.py:726]: I did a quick unit test and didn't see any problem.
`log_router.stream_logs` also defines its stream generator outside of its endpoint.

Comment (ogabrielluiz) [reply src/backend/base/langflow/api/v1/chat.py:726]: Ran unit tests and it worked.

**Review by ogabrielluiz — APPROVED** [review]

**Pull request merged**